### PR TITLE
[beta] UX: slow legacy pagination needs loading spinners

### PR DESCRIPTION
**Pull request by MilanPospisil**
Issue: AAH-2062
https://issues.redhat.com/browse/AAH-2062

Correct spinner in Legacy Roles and Namespaces.

### CODE REVIEW

File: src/containers/legacy-namespaces/legacy-namespaces.tsx [hunk 1]
```diff
@@ -83,24 +83,26 @@ class LegacyNamespaces extends React.Component<
...
+          params: {
+            page: page,
+            page_size: page_size,
+            order_by: order_by,
+            keywords: keywords,
+          },
+          count: response.data.count,
+          legacynamespaces: response.data.results,
+        }));
+      });
```
Comment (himdel) [inline src/containers/legacy-namespaces/legacy-namespaces.tsx:105]: the change looks good, but the same code exists in componentDidMount as well .. move it to a query(params) methods you can call from both?

File: src/containers/legacy-roles/legacy-roles.tsx [hunk 2]
```diff
@@ -87,26 +87,28 @@ class LegacyRoles extends React.Component<RouteComponentProps, IProps> {
...
+            page: page,
+            page_size: page_size,
+            order_by: order_by,
+            keywords: keywords,
+            tags: tags,
+          },
+          count: response.data.count,
+          legacyroles: response.data.results,
+        }));
+      });
```
Comment (himdel) [inline src/containers/legacy-roles/legacy-roles.tsx:111]: (and here as well)

**Review by himdel — APPROVED** [review]
Loading spinners appear on load, pagination and filtering now :+1: 

(Empty states on filtering are missing/broken, and updateParams doesn't seem to be updating params, but those are unrelated.)

File: src/containers/legacy-namespaces/legacy-namespaces.tsx (hunk 1 above)
Comment (MilanPospisil) [reply src/containers/legacy-namespaces/legacy-namespaces.tsx:105]: Yes, that was my first though, but then I heard your voice: "Dont do more work than Issue says" :-D. I can do it now.

File: src/containers/legacy-roles/legacy-roles.tsx [hunk 3]
```diff
@@ -54,59 +54,42 @@ class LegacyRoles extends React.Component<RouteComponentProps, IProps> {
...
-        legacyroles: response.data.results,
-      }));
-    });
+    p.page = parseInt(urlParams.get('page')) || 1;
+    p.page_num = parseInt(urlParams.get('page')) || 1;
+    p.page_size = parseInt(urlParams.get('page_size')) || 10;
+    p.order_by = urlParams.get('order_by') || 'created';
+    p.keywords = urlParams.get('keywords');
+    p.tags = urlParams.get('tags');
+    this.updateParams(p);
```
Comment (himdel) [inline src/containers/legacy-roles/legacy-roles.tsx:66]: Looks good, except for that `p: any`..

(also, page_num becomes unused)
```suggestion
    this.updateParams({
      page: parseInt(urlParams.get('page'), 10) || 1,
      page_size: parseInt(urlParams.get('page_size'), 10) || 10,
      order_by: urlParams.get('order_by') || 'created',
      keywords: urlParams.get('keywords'),
      tags: urlParams.get('tags'),
    });
```

File: src/containers/legacy-namespaces/legacy-namespaces.tsx [hunk 4]
```diff
@@ -53,14 +53,16 @@ class LegacyNamespaces extends React.Component<
...
-    const page_size = parseInt(urlParams.get('page_size')) || 10;
-    const order_by = urlParams.get('order_by') || 'name';
-    const keywords = urlParams.get('keywords') || null;
+    p.page = parseInt(urlParams.get('page')) || 1;
+    p.page_size = parseInt(urlParams.get('page_size')) || 10;
+    p.order_by = urlParams.get('order_by') || 'name';
+    p.keywords = urlParams.get('keywords') || null;
 
-    LegacyNamespaceAPI.list({
+    this.updateParams(p);
```
Comment (himdel) [inline src/containers/legacy-namespaces/legacy-namespaces.tsx:64]:
```suggestion
    this.updateParams({
      page: parseInt(urlParams.get('page'), 10) || 1,
      page_size: parseInt(urlParams.get('page_size'), 10) || 10,
      order_by: urlParams.get('order_by') || 'name',
      keywords: urlParams.get('keywords') || null,
    });
```

and remove the commented out code please :)

File: src/containers/legacy-namespaces/legacy-namespaces.tsx (hunk 1 above)
Comment (himdel) [reply src/containers/legacy-namespaces/legacy-namespaces.tsx:107]: Heh, and now you see why :)

File: src/containers/legacy-namespaces/legacy-namespaces.tsx [hunk 5]
```diff
@@ -55,52 +55,36 @@ class LegacyNamespaces extends React.Component<
...
-          page: page,
-          page_size: page_size,
-          order_by: order_by,
-          keywords: keywords,
-        },
-        count: response.data.count,
-        legacynamespaces: response.data.results,
-      }));
+    this.updateParams({
+      page: parseInt(urlParams.get('page'), 1) || 1,
```
Comment (himdel) [inline src/containers/legacy-namespaces/legacy-namespaces.tsx:59]:
```suggestion
      page: parseInt(urlParams.get('page'), 10) || 1,
```

base 1 is ..problematic :)

Comment (himdel) [inline src/containers/legacy-namespaces/legacy-namespaces.tsx:59]:
```suggestion
      page: parseInt(urlParams.get('page'), 10) || 1,
```

base 1 is ..problematic :)

File: src/containers/legacy-roles/legacy-roles.tsx [hunk 6]
```diff
@@ -56,57 +56,39 @@ class LegacyRoles extends React.Component<RouteComponentProps, IProps> {
...
-          page: page_num,
-          page_size: page_size,
-          order_by: order_by,
-          keywords: keywords,
-        },
-        count: response.data.count,
-        legacyroles: response.data.results,
-      }));
+    this.updateParams({
+      page: parseInt(urlParams.get('page'), 1) || 1,
```
Comment (himdel) [inline src/containers/legacy-roles/legacy-roles.tsx:60]: (and here, parseInt should work with base 10)

Comment (MilanPospisil) [reply src/containers/legacy-roles/legacy-roles.tsx:60]: Ah, this is the radix, I though this is default value when string is null (and I wondered why is there another default value after it). And I wonder why radix is not always default to 10, when empty... nonconsistent browsers?

**Comment by himdel** [comment]
Now all of them are have radix 1 instead of 10 ;)

**Review by himdel — APPROVED** [review]

**Pull request merged**